internal/mapping.go

```diff
@@ -30,23 +30,29 @@ import (
 // a) the new table name is legal
 // b) the new table name doesn't clash with other Spanner table names
 // c) we consistently return the same name for this table.
+//
+// conv.UsedNames tracks Spanner names that have been used for table names, foreign key constraints
+// and indexes. We use this to ensure we generate unique names when
+// we map from MySQL to Spanner since Spanner requires all these names to be
+// distinct and should not differ only in case.
 func GetSpannerTable(conv *Conv, srcTable string) (string, error) {
 	if srcTable == "" {
 		return "", fmt.Errorf("bad parameter: table string is empty")
 	}
+	// Once computed, return cached result.
 	if sp, found := conv.ToSpanner[srcTable]; found {
 		return sp.Name, nil
 	}
 	spTable, _ := FixName(srcTable)
-	if _, found := conv.ToSource[spTable]; found {
+	if _, found := conv.UsedNames[strings.ToLower(spTable)]; found {
```

Comment:
can we re-use the getSpannerId method here ?

Reply:
Not directly, but moved the common code to a new function